enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,257 @@
+    1. Delete `prometheus-adapter` resources
+
+### API Extensions
+
+No new API extensions are required. For tech preview, the only update required
+is the addition of a `MetricsServer` feature gate to the OpenShift API
+`TechPreviewNoUpgrade` feature set. See [Design Details](#design-details) for
+the specifics.
+
+### Implementation Details/Notes/Constraints
+
+#### Impact for Hosted Control Planes (HyperShift)
+
+1. The existing resource metrics API implementation, prometheus-adapter, is deployed on the guest cluster's data plane by CMO which is itself deployed by the Cluster Version operator. The same will be true for the Metrics server because it requires kubelet scraping. Consequently, we don't anticipate any effects on Hypershift as part of this enhancement.
+
+#### Configuration/Enablement
+
+Initially the Metrics Server must be explicitly enabled by the user by configuring the `TechPreviewNoUpgrade` feature set. 
+
+#### Resource Impact
+
+Once metrics-server is enabled, by default 2 replicas of `metrics-server` instances are being deployed for for multi-nodes clusters and 1 for SNO clusters in the openshift-monitoring namespace.
+
+Once metrics-server is fully functional (after 5 scrape cycles done by metrics-server) we will remove `prometheus-adapter` resources fully.
+
+### Risks and Mitigations
+
+With prometheus-adapter only Prometheus scrapes kubelet, but with metrics-server you need both Prometheus and metrics-server to scrape the kubelet in order to also get container metrics in Prometheus.
+
+We compared cpu and memory usage of metric-server vs prometheus-adapter on cluster-bot's cluster in [MON-3210](https://issues.redhat.com/browse/MON-3210) which didn't see significant difference when prometheus-adapter was replaced with metrics-server.
+
+We need to do performance testing on a larger OpenShift cluster (more than 100 nodes) and more than 70 pods on per node to measure the performance of Metrics Server. This is based on scalability table mentioned in [metrics-server](https://github.com/kubernetes-sigs/metrics-server#scaling). [MON-3328](https://issues.redhat.com/browse/MON-3328) is created for doing this.
+
+### Drawbacks
+
+N/A
+
+
+## Design Details
+
+We will add a new feature gate `MetricsServer` which will be enabled by default in the `TechPreviewNoUpgrade` feature set.
+
+For the cluster monitoring operator, we propose to add a type `metricsServer` to the `openshift-montoring/cluster-monitoring-config` configmap defining the
+metrics-server settings such as pod placement and resource requirements.
+
+  ```go
+  type ClusterMonitoringConfiguration struct {
+    ...
+    // `MetricsServerConfig` defines settings for the MetricsServer component.
+    MetricsServerConfig *MetricsServerConfig `json:"metricsServer,omitempty"`
+  }
+
+  // The `MetricsServerConfig` resource defines settings for the MetricsServer component.
+  type MetricsServerConfig struct {
```

Comment:
sorry I missed it in previous reviews but we have a few moe prometheus adapter settings that we need to port.
```suggestion
  type MetricsServerConfig struct {
    Audit *Audit `json:"audit,omitempty"`
```